crates/wasm-encoder/src/component/aliases.rs

```diff
@@ -106,10 +106,11 @@ impl ComponentAliasSection {
     }
 
     /// Define an alias to a component instance's export.
-    pub fn alias(&mut self, alias: Alias<'_>) -> &mut Self {
+    pub fn alias(&mut self, alias: Alias<'_>) -> u32 {
         alias.encode(&mut self.bytes);
+        let index = self.num_added;
         self.num_added += 1;
-        self
+        index
```

Comment:
This is the index within this particular alias section, which is not the same thing as an item index as there can be multiple alias sections.

Additionally, alias indexes don't make sense as the new index depends on what is being aliased.